common/common.go

```diff
@@ -561,6 +561,13 @@ type NotificationInfo struct {
 	MetaData          *MetaData
 }
 
+type ObjectInQueue struct {
+	NotificationAction string // Notification status and type
+	NotificationType   string
+	Object             *MetaData
```

Comment:
Since you're passing a ptr, make sure that no code modifies the metadata object after it is put into the channel.